sherpa/ui/utils.py

```diff
@@ -10664,15 +10667,11 @@ def get_data_plot_prefs(self, id=None):
 
         """
 
+        plot = self.get_data_plot(id, recalc=False)
         try:
-            d = self.get_data(id)
-            if isinstance(d, sherpa.data.Data1DInt):
-                return self._datahistplot.histo_prefs
-
-        except IdentifierErr:
-            pass
-
-        return self._dataplot.plot_prefs
+            return plot.histo_prefs
+        except AttributeError:
```

Comment:
Is the more common case here typically the dataset having the histo_prefs or raising the Attribute error? If we expect the latter, perhaps using hasattr() might be more effiicient.

Reply:
Either way, this is not performance-critical. It's something that will come up no more than a dozen times (no more than a few time per plot if someone runs a script to automatically generate a lot of plots) in a typical few-hour analysis session and probably save < 0.1 ms each time. Discussing this might well take longer than the potential performance gain for all sherpa users ever combined ;-)

Reply:
@wmclaugh - there is a cottage industry on saying whether to use hasattr or not - e.g. https://hynek.me/articles/hasattr/